### ISSUE TITLE
Monitoring and moderation

### ISSUE THREAD

**Issue by dominijk** [tomalrussell/colouring-london#284]
**The alert button**

If someone hits the alter (panic button) then this should immediately hide the last edit and set in place actions to investigate. The type of investigation can link to the reason for pressing the button. How does all this work? Who is monitoring and who is moderating. 

**URL / free text checking**
Where we allow links then they should undergo a check before being visible for new members, once a member is established and trusted maybe this is less necessary? The check could be something like;

`IF` URL given;
1.  `THEN` URL checked against a register of known trusted sites (compiled from those within our system) if trusted site goes,
2. `ELSE` if not check against a dictionary of malicious sites (are there open registers of these?) if on that dictionary delete and flag user (what do we do with users previously entered data)
3. `ELSE IF` ping URL and check it's live and valid if not live then remove (and notify user link wasn't parsed) 
4. `ELSE IF` none of the above and valid link pass to a moderator for human review

**Comment by dominijk** [comment]
Updated following design review, alert button not required by controlling free text boxes.

**Comment by polly64** [comment]
Dom and I had along chat about this last week.  We agreed the priority was to
a) avoid freetext options wherever possible 
b) where freetext necessary include dropdown options - also adds layer of complexity to process to deter people
c) to gradually use entries to build up a single dropdown option lists for all sources - for example in the end to be able to choose book start typing Pevsner and then it will appear and you can select it and then just put numbers in for chapter and page so we're just putting in numbers wherever we can..
d) Block text and weblink release before moderation
Is possible to give me an easy way to check the stuff? I can take this on for next couple of months until we can advance dropdown lists (I'm also working on expert moderation groups) 

Also are we planning to use
a) open software able to check malicious weblinks?
b) Open software checking for malicious words

**Comment by tomalrussell** [comment]
I agree with a) "avoid freetext where possible" and b) "include dropdown options" - these are sensible and simple policies to adopt.

I think c) "building up suggestions" sounds useful but worth noting that this is a large feature - need to track and make source suggestions, introduce a data model for citations, develop the UI for data entry, validation for all of this.

I disagree with d) "block before moderation". I think we should use "retrospective review" (https://collaborative-data.theodi.org/patterns/maintaining-quality/retrospective-review) and introduce features to:
- make it easy to see recent edits
- easily revert changes
- flag changes for review (without needing to login to raise a flag)
- reject changes which include blacklisted malicious words or URLs (backend validation)

Overall this is simpler than holding changes for moderation, and encourages non-malicious users (who are hopefully the majority! and definitely the most important to us) by accepting their changes immediately. It also makes community review possible - anyone can immediately fix something they spot, whether it's malicious or simply incorrect.

**Comment by polly64** [comment]
thanks Tom all sounds good

**Comment by mz8i** [comment]
#499 created to fulfil the "make it easy to see recent edits" requirement mentioned above

**Comment by polly64** [comment]
Just thinking again this blocking issue as I think needed for specific security cases until we have things in place. Let's discuss more might be easier on slack will add thoughts now
